src/x509.c

```diff
@@ -5444,63 +5483,62 @@ static int X509PrintSerial(WOLFSSL_BIO* bio, WOLFSSL_X509* x509, int indent)
     unsigned char serial[32];
     int  sz = sizeof(serial);
     char scratch[MAX_WIDTH];
+    int scratch_len;
 
     XMEMSET(serial, 0, sz);
     if (wolfSSL_X509_get_serial_number(x509, serial, &sz) == WOLFSSL_SUCCESS) {
-
-        XSNPRINTF(scratch, MAX_WIDTH, "%*s%s", indent, "",
-                "Serial Number:");
-        if (wolfSSL_BIO_write(bio, scratch, (int)XSTRLEN(scratch)) <= 0) {
+        if ((scratch_len = XSNPRINTF(scratch, MAX_WIDTH, "%*sSerial Number:",
+                                     indent, "")) >= MAX_WIDTH)
+        {
+            WOLFSSL_MSG("buffer overrun");
+            return WOLFSSL_FAILURE;
+        }
+        if (wolfSSL_BIO_write(bio, scratch, scratch_len) <= 0) {
             return WOLFSSL_FAILURE;
         }
 
         if (sz > (int)sizeof(byte)) {
             int i;
-            char tmp[100];
-            int  tmpSz = 100;
-            char val[5];
-            int  valSz = 5;
+            int val_len;
 
             /* serial is larger than int size so print off hex values */
-            XSNPRINTF(scratch, MAX_WIDTH, "\n%*s", indent, "");
-            if (wolfSSL_BIO_write(bio, scratch, (int)XSTRLEN(scratch))
-                    <= 0) {
+            if ((scratch_len = XSNPRINTF(
+                     scratch, MAX_WIDTH, "\n%*s", indent, ""))
+                    >= MAX_WIDTH)
+            {
```

Comment:
{ on same line with if statement

Reply:
(done -- not needed due to ad-hoc spacing here carried forward from incumbent code.)